### PR TITLE
DRY up net.go

### PR DESCRIPTION
**Pull request by rodentrabies**
In response to #214. Abstacted out process of sending message to peer given by peer ID. Total of -113 lines of code, which is good, but I'm not sure it is really useful, as it makes code in those functions non-symmetrical and, as a result, harder to mentally parse as a whole (similar to @cpacia's note at #217). On the other hand, less code == less space to make something wrong, so either close or merge, depending on what is more suitable for the goals of the project.

### CODE REVIEW

File: core/net.go [hunk 1]
```diff
@@ -61,27 +61,30 @@ func (n *OpenBazaarNode) SendOfflineMessage(p peer.ID, m *pb.Message) error {
...
 
+func (n *OpenBazaarNode) SendOfflineAck(peerId string, pointerID peer.ID) error {
+	message := pb.Message{
+		MessageType: pb.Message_OFFLINE_ACK,
+		Payload:     &any.Any{Value: []byte(pointerID.Pretty())},
+	}
+	return n.peerIDSendMessage(peerId, message)
+}
+
 func (n *OpenBazaarNode) GetPeerStatus(peerId string) string {
```
Comment (duomoney) [inline core/net.go]: Should this function be pushed up?

**Comment by duomoney** [comment]
What do you mean by "it makes code in those functions non-symmetrical"? Are you referring to `SendOrder` being exceptional?

File: core/net.go [hunk 2]
```diff
@@ -158,142 +136,51 @@ func (n *OpenBazaarNode) SendOrder(peerId string, contract *pb.RicardianContract
 }
 
 func (n *OpenBazaarNode) SendOrderConfirmation(peerId string, contract *pb.RicardianContract) error {
-	p, err := peer.IDB58Decode(peerId)
-	if err != nil {
-		return err
-	}
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
 	any, err := ptypes.MarshalAny(contract)
```
Comment (duomoney) [inline core/net.go]: For consistency, rename the `any` variables to `a`, or vice-versa.

Comment (yurizhykin) [inline core/net.go]: Done

**Comment by yurizhykin** [comment]
@duomoney, yes, exactly that

**Comment by JustinDrake** [comment]
@yurizhykin I find it strange that `SendOrder` uses `SendRequest` but the other functions use `SendMessage` or `SendOfflineMessage`. This is for two reasons:

1) It seems to suggest that `SendOrder` doe not work in the offline case. I though offline orders was part of OB 2.0.
2) It seems to suggest that the response for the actions other than `SendOrder` are not worth looking at (e.g. for errors). Error checking here seems to be missing.

Maybe @cpacia can clarify.

**Comment by jackkleeman** [comment]
@JustinDrake In the offline case, SendOrder returns an error, at which point Purchase() resorts to using SendOfflineMessage. Why it is done this way (outside of net.go) unlike all other messaging I am not sure.

**Comment by cpacia** [comment]
If the vendor is offline it changes the data we need to send so it doesn't work to just send the message as-is as an offline message. We need to modify it before sending the offline message.

**Comment by cpacia** [comment]
I fixed the merge conflicts and and bugs created by other changes to net and merge.

**Pull request closed without merging**